src/Stack/Options.hs

```diff
@@ -682,7 +683,12 @@ initOptsParser :: Parser InitOpts
 initOptsParser =
     InitOpts <$> solver <*> omitPackages
              <*> overwrite <*> fmap not ignoreSubDirs
+             <*> searchDirs
   where
+    searchDirs =
+      many (textArgument
+              (metavar "SEARCH-DIRECTORIES" <>
+               help "Directories which are searched for cabal files. If non specified, uses current directory."))
```

Comment:
I can change it to be more concise. I was going off what @mgsloan suggested before. But I do like the more concise version better.